support/routes/v2/isobot/ops/botService.ts

```diff
@@ -91,6 +170,276 @@ class BotService {
 
     await this.whitelistService.addWhitelist(emails)
   }
+
+  getValidatedDomain(payload: SlackPayload) {
+    const { text: domain } = payload
+    const DOMAIN_NAME_REGEX = /^[a-zA-Z0-9][a-zA-Z0-9-]{0,61}[a-zA-Z0-9](?:\.[a-zA-Z]{2,})+$/
+
+    if (!DOMAIN_NAME_REGEX.test(domain)) {
+      return false
+    }
+
+    return domain
+  }
+
+  getDnsCheckerMessage(
+    domain: string,
+    cnameDomain: string | null,
+    redirectionDomain: string,
+    cnameRecord: string | null,
+    indirectionDomain: string,
+    intermediateRecords: string[] | null,
+    redirectionRecords: string[] | null
+  ) {
+    // Domain has a CNAME pointing to one of our known suffixes
+    const isDomainCnameCorrect =
+      !!cnameRecord &&
+      (cnameRecord.endsWith(`.${DNS_INDIRECTION_DOMAIN}`) ||
+        DNS_CNAME_SUFFIXES.some((suffix) => cnameRecord.endsWith(`.${suffix}`)))
+
+    // Domain is directly pointing to our indirection layer
+    const isDomainOnIndirection =
+      cnameRecord && cnameRecord.endsWith(`.${DNS_INDIRECTION_DOMAIN}`)
+
+    // The intermediate layer (indirection or CNAME) is resolving to valid IPs
+    // If on KeyCDN, then there should only be 1 IP, otherwise there should be 4
+    const isIntermediateValid =
+      intermediateRecords &&
+      ((cnameRecord &&
+        cnameRecord.endsWith(".kxcdn.com") &&
```

Comment:
i think we declared the suffixes in a different `constants.ts` file

Reply:
https://github.com/isomerpages/isomercms-backend/pull/1367#discussion_r1599217406